File: scala/trip_service_kata/src/main/scala/org/craftedsw/tripservicekata/trip/TripDAO.scala
package org.craftedsw.tripservicekata.trip

import org.craftedsw.tripservicekata.user.User
import org.craftedsw.tripservicekata.exception.CollaboratorCallException

object TripDAO extends TripDAO {

	def findTripsByUser(user: User): List[Trip] = {
		throw new CollaboratorCallException(
			"TripDAO should not be invoked on an unit test.");
	}

}

trait TripDAO {

	def findTripsByUser(user: User): List[Trip]

}

File: Objective-C/TripServiceKata/TripServiceKata/TripDAO.m
#import "TripDAO.h"
#import "DependendClassCallDuringUnitTestException.h"

@implementation TripDAO

+(NSArray*)findTripsByUser:(User*)user
{
    [DependendClassCallDuringUnitTestException raise:@"Unit Test Exception" format:@"TripDAO should not be invoked on an unit test."];
    
    return nil;
}



@end
